source/guidelines/sdk-development.rst

```diff
@@ -281,14 +212,30 @@ KeyPair and Cryptographic functions
 
 Implementing cryptographic functions is required to sign transactions.
 
-Example: `core/crypto <https://github.com/nemtech/nem2-sdk-java/tree/master/sdk-core/src/main/java/io/nem/core/crypto>`_
+Example: `core/crypto <https://github.com/nemtech/symbol-sdk-typescript-javascript/tree/master/src/core/crypto>`_
```

Comment:
Creators of SDKs should pay special attention to the [vector](https://github.com/nemtech/test-vectors) test. They should create unit tests using the files as input and expected outputs.

Examples of vector:

- https://github.com/nemtech/symbol-sdk-java/blob/master/sdk-core/src/test/java/io/nem/symbol/core/crypto/KeyPairVectorTester.java
- https://github.com/nemtech/symbol-sdk-java/blob/master/sdk-core/src/test/java/io/nem/symbol/core/crypto/DsaSignerVectorTester.java
- https://github.com/nemtech/symbol-sdk-typescript-javascript/blob/master/test/core/crypto/keyPair.spec.ts#L88

Can we extend this section with the vector tests?

Reply:
Regarding the recent server updates:
- SDKs are now using standard `tweetnacl` (ed2559) for keyPair generation, address derivation (from public key) and signings. Keypairs are based on `tweetnacl` 64 bytes secretKey (public + private) using `sha512`. 
- Signatures also get generated using sha512. In the latest SDK updates we use `tweetnacl` detached mode for signing. 
- Please be noded that we are still using a custom script for `SharedKey` derivation in the SDK. This is due to the ED2559 Mult function (Edward Curve) is used as an internal function which is not exposed in `tweetnacl`.  There's a discussion around this topic whether to change on the server to adapt the exposed mult function. So, keep eyes on future announcement.